### PR TITLE
feat: migrate to rspress

### PR DESCRIPTION
**Pull request by sanyuan0704**
Migrate oxc website to Rspress, reasons:

- Less configuration.
- Provide full-text search out of the box.
- Performance. Thanks for the power of Oxc and Rspack, for the the website project, startup time in dev  < 1s,  build time in production < 1s, which VitePress can't reach.

image

### CODE REVIEW

File: docs/learn/ecmascript/spec.md [hunk 1]
```diff
@@ -3,6 +3,10 @@ id: ecma_spec
 title: Specification
 ---
 
+ß
```
Comment (chenjiahan) [inline docs/learn/ecmascript/spec.md:6]: Is this a invalid char?

File: package.json [hunk 2]
```diff
@@ -3,33 +3,20 @@
...
-    "start": "docusaurus start",
-    "build": "docusaurus build",
-    "swizzle": "docusaurus swizzle",
-    "deploy": "docusaurus deploy",
-    "clear": "docusaurus clear",
-    "serve": "docusaurus serve",
-    "write-translations": "docusaurus write-translations",
-    "write-heading-ids": "docusaurus write-heading-ids",
+    "start": "rspress start",
+    "build": "rspress build",
```
Comment (chenjiahan) [inline package.json:7]: We can `rspress preview` command here

**Comment by Boshen** [comment]
Thank you for working on this! I made a vote since we have 2 PRs https://twitter.com/boshen_c/status/1723535140343931205

**Comment by sanyuan0704** [comment]
今天的局面是我不情愿看见的，我很反感这种对于这种不经开发者同意就挑起竞争和冲突的行为。

**Pull request closed without merging**